core-to-plc/test/Spec.hs

```diff
@@ -9,26 +9,40 @@ module Main (main) where
 import           IllTyped
 
 import           Language.Plutus.CoreToPLC.Plugin
-import           Language.Plutus.CoreToPLC.Primitives as Prims
+import qualified Language.Plutus.CoreToPLC.Primitives     as Prims
 
 import           Language.PlutusCore
+import           Language.PlutusCore.Evaluation.CkMachine
 
 import           Test.Tasty
 import           Test.Tasty.Golden
 
 import           Control.Exception
-import qualified Data.ByteString.Lazy                 as BSL
-import           Data.Text                            as T
-import           Data.Text.Encoding                   (encodeUtf8)
+import           Control.Monad.Except                     hiding (void)
+
+import qualified Data.ByteString.Lazy                     as BSL
+import qualified Data.Text                                as T
+import           Data.Text.Encoding                       (encodeUtf8)
 
 -- this module does lots of weird stuff deliberately
 {-# ANN module "HLint: ignore" #-}
 
 main :: IO ()
 main = defaultMain tests
 
+getPlc :: PlcCode -> ExceptT BSL.ByteString IO (Program TyName Name ())
+getPlc value = withExceptT (strToBs . show) $ getAst <$> (ExceptT $ try @SomeException (evaluate value))
```

Comment:
I think I'm happy to leave it in this case (since what will happen in the case that you describe will cause a test failure), but I will read up on `safe-exceptions` for the future (I have been trying to avoid learning the horrible details of Haskell's exceptions...)